src/utils/context.rs

```diff
@@ -17,7 +17,6 @@ impl<'r> Context<'r> {
         self.client.inner()
     }
 
-    // TODO: we would want to return the Cache trait here
```

Comment:
Why do we remove this TODO? it is not solved, right?

Reply:
I removed it because of the conversation with Nick in the second pair programming session. We can add the TODO back in if you prefer. If you want I can look into the recording and point you to the timestamp so you can have a listen. 

I remember it was something about `ServiceCache` already wrapping the implementation of cache. If you look at: https://github.com/gnosis/safe-client-gateway/blob/48a957fdcf80d9986a20dcbb1a0da9aa60008854/src/cache/redis.rs#L11

We could probably introduce that cache implementation decision logic there, more easily when necessary.

Reply:
We could just have a different `State<'r, SomeOtherCacheType>`